python/lsst/sims/photUtils/Sed.py

```diff
@@ -81,14 +81,253 @@
+                               + "this is for security reasons\n"
+                               + "https://docs.python.org/3.1/library/pickle.html#pickle-restrict")
+
+        if module == "numpy":
+            if name == "ndarray":
+                return getattr(numpy, name)
+            elif name == "dtype":
+                return getattr(numpy, name)
+            else:
+                raise RuntimeError("sed_unpickler not meant to load numpy.%s" % name)
+        elif module == "numpy.core.multiarray":
+            return getattr(numpy.core.multiarray, name)
+        else:
+            raise RuntimeError("sed_unpickler cannot handle module %s" % module)
+
+
+def _validate_sed_cache():
+    """
+    Verifies that the pickled SED cache exists, is a dict, and contains
+    an entry for every SED in starSED/ and galaxySED.  Does nothing if so,
+    raises a RuntimeError if false.
+
+    We are doing this here so that sims_sed_library does not have to depend
+    on any lsst testing software (in which case, users would have to get
+    a new copy of sims_sed_library every time the upstream software changed).
+
+    We are doing this through a method (rather than giving users access to
+    _global_lsst_sed_cache) so that users do not accidentally ruin
+    _global_lsst_sed_cache.
+    """
+    global _global_lsst_sed_cache
+    if _global_lsst_sed_cache is None:
+        raise SedCacheError("_global_lsst_sed_cache does not exist")
+    if not isinstance(_global_lsst_sed_cache, dict):
+        raise SedCacheError("_global_lsst_sed_cache is a %s; not a dict"
+                           % str(type(_global_lsst_sed_cache)))
+    sed_dir = getPackageDir('sims_sed_library')
+    sub_dir_list = ['galaxySED', 'starSED']
+    file_ct = 0
+    for sub_dir in sub_dir_list:
+        tree = os.walk(os.path.join(sed_dir, sub_dir))
+        for entry in tree:
+            local_dir = entry[0]
+            file_list = entry[2]
+            for file_name in file_list:
+                if file_name.endswith('.gz'):
+                    full_name = os.path.join(sed_dir, sub_dir, local_dir, file_name)
+                    if full_name not in _global_lsst_sed_cache:
+                        raise SedCacheError("%s is not in _global_lsst_sed_cache"
+                                           % full_name)
+                    file_ct += 1
+    if file_ct == 0:
+        raise SedCacheError("There were not files in _global_lsst_sed_cache")
+
+    return
+
+def _compare_cached_versus_uncached():
+    """
+    Verify that loading an SED from the pickled cache give identical
+    results to loading the same SED from ASCII
+    """
+    sed_dir = os.path.join(getPackageDir('sims_sed_library'),
+                           'starSED', 'kurucz')
+
+    dtype = numpy.dtype([('wavelen', float), ('flambda', float)])
+
+    sed_name_list = os.listdir(sed_dir)
+    msg = ('An SED loaded from the pickled cache is not '
```

Comment:
should `msg` include instructions on how to re-build the cache if there is an error?